File: weather/src/main/java/com/github/xiaogegechen/weather/view/impl/SettingActivity.java
package com.github.xiaogegechen.weather.view.impl;

import android.widget.ListView;
import android.widget.Switch;

import com.github.xiaogegechen.common.base.BaseActivity;
import com.github.xiaogegechen.common.util.StatusBarUtils;
import com.github.xiaogegechen.common.util.ToastUtil;
import com.github.xiaogegechen.design.viewgroup.TitleBar;
import com.github.xiaogegechen.design.viewgroup.impl.OnArrowClickListenerAdapter;
import com.github.xiaogegechen.weather.R;
import com.github.xiaogegechen.weather.adapter.SettingAdapter;
import com.github.xiaogegechen.weather.model.Setting;
import com.github.xiaogegechen.weather.presenter.ISettingActivityPresenter;
import com.github.xiaogegechen.weather.presenter.impl.SettingActivityPresenterImpl;
import com.github.xiaogegechen.weather.view.ISettingActivityView;

import java.util.ArrayList;
import java.util.List;

public class SettingActivity extends BaseActivity implements ISettingActivityView {
    private TitleBar mTitleBar;
    private ListView mListView;

    private ISettingActivityPresenter mSettingActivityPresenter;

    @Override
    public void initView() {
        mTitleBar = findViewById(R.id.titleBar);
        mListView = findViewById(R.id.listView);
        // imm
        StatusBarUtils.setImmersive(this);
        StatusBarUtils.fillStatusBarByView(
                this,
                findViewById(R.id.root),
                getResources().getColor(R.color.weather_color_primary)
        );
    }

    @Override
    public void initData() {
        mSettingActivityPresenter = new SettingActivityPresenterImpl();
        mSettingActivityPresenter.attach(this);
        // mTitleBar
        mTitleBar.setListener(new OnArrowClickListenerAdapter(){
            @Override
            public void onLeftClick() {
                finish();
            }
        });
        // mListView
        mListView.setAdapter(new SettingAdapter(
                this,
                R.layout.weather_activity_setting_item,
                getListViewDataSource()
        ));
        mListView.setOnItemClickListener((parent, view, position, id) -> {
            // 点击事件
            switch (position){
                case 0:
                    // 允许背景图切换
                    Switch switchButton = view.findViewById(R.id.switchButton);
                    boolean newState = !switchButton.isChecked();
                    mSettingActivityPresenter.setAllowBgChange(newState);
                    switchButton.setChecked(newState);
                    break;
                default:
                    break;
            }
        });
    }

    private List<Setting> getListViewDataSource(){
        List<Setting> dataSource = new ArrayList<>();
        // 允许背景图切换
        boolean allowBgChange = mSettingActivityPresenter.isAllowBgChange();
        dataSource.add(new Setting(getString(R.string.weather_setting_allow_bg_change), true, allowBgChange));
        return dataSource;
    }

    @Override
    protected void onDestroy() {
        mSettingActivityPresenter.detach();
        super.onDestroy();
    }

    @Override
    public int getLayoutId() {
        return R.layout.weather_activity_setting;
    }

    @Override
    public boolean isSupportSwipeBack() {
        return false;
    }

    @Override
    public void showProgress() {
    }

    @Override
    public void showErrorPage() {
    }

    @Override
    public void showToast(String message) {
        ToastUtil.show(message);
    }
}

File: module_left/src/main/java/com/github/xiaogegechen/module_left/view/SettingActivity.kt
package com.github.xiaogegechen.module_left.view

import android.os.Bundle
import android.widget.ListView
import android.widget.Switch
import com.github.xiaogegechen.common.base.BaseActivity
import com.github.xiaogegechen.design.TitleBar
import com.github.xiaogegechen.module_left.R
import com.github.xiaogegechen.module_left.adapter.SettingAdapter
import com.github.xiaogegechen.module_left.model.Setting
import com.github.xiaogegechen.module_left.presenter.SettingActivityPresenterImpl
import java.util.ArrayList

class SettingActivity : BaseActivity(), ISettingActivityView {

    private var mTitleBar: TitleBar? = null
    private var mListView: ListView? = null

    private var mSettingActivityPresenterImpl: SettingActivityPresenterImpl? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mTitleBar = findViewById(R.id.settingTitleBar)
        mListView = findViewById(R.id.settingListView)

        mListView?.adapter = SettingAdapter(getContext()!!, R.layout.module_left_setting_activity_list_item, DATA)

        mTitleBar?.setListener(object: TitleBar.OnArrowClickListener{
            override fun onLeftClick() {
                finish()
            }

            override fun onRightClick() {
            }
        })

        mListView?.setOnItemClickListener{ _, view, position, _ ->
            val setting = DATA[position]
            when(setting.name){
                "应用信息" -> mSettingActivityPresenterImpl?.gotoInformationActivity()
                "清理缓存" -> mSettingActivityPresenterImpl?.clearCache()
                "自动更新" -> {
                    view.findViewById<Switch>(R.id.setSwitch).isChecked = !view.findViewById<Switch>(R.id.setSwitch).isChecked
                    mSettingActivityPresenterImpl?.setAutoRefreshEnable(view.findViewById<Switch>(R.id.setSwitch).isChecked)
                }
                "允许前台推送" ->{
                    view.findViewById<Switch>(R.id.setSwitch).isChecked = !view.findViewById<Switch>(R.id.setSwitch).isChecked
                    mSettingActivityPresenterImpl?.setShowNotificationEnable(view.findViewById<Switch>(R.id.setSwitch).isChecked)
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        mSettingActivityPresenterImpl?.detach()
    }

    override fun showProgress() {
    }

    override fun showErrorPage() {
    }

    override fun showToast(message: String) {
    }

    override fun initData() {
        mSettingActivityPresenterImpl = SettingActivityPresenterImpl()
        mSettingActivityPresenterImpl?.attach(this)
    }

    override fun getLayoutId(): Int {
        return R.layout.module_left_activity_setting
    }

    override fun getStatusBarColor(): Int {
        return resources.getColor(R.color.design_color_accent)
    }

    override fun isSupportSwipeBack(): Boolean {
        return true
    }

    companion object{
        private val DATA: MutableList<Setting>
        init {
            DATA = ArrayList()
            DATA.add(Setting("应用信息", false))
            DATA.add(Setting("清理缓存", false))
            DATA.add(Setting("自动更新", true))
            DATA.add(Setting("允许前台推送", true))
        }
    }

}
